src/pac_array.cc

```diff
@@ -252,7 +252,9 @@ void ArrayType::GenArrayLength(Output* out_cc, Env* env, const DataPtr& data)
 	if ( ! incremental_parsing() )
 		{
 		arraylength_var_field_->GenTempDecls(out_cc, env);
-		arraylength_var_field_->GenInitCode(out_cc, env);
+		// This is about to get initialized below, don't initialize it twice.
```

Comment:
Uhm ... where? Just going on pattern-matching I don't immediately see it.

Reply:
It adds a second initialization for `arraylength_var_field` in the next `if` block in the file. Prior to this change, binpac would generate code that looks like this:

```
	int t_key_object__arraylength;
	t_key_object__arraylength = 0;
	t_key_object__arraylength = key_object_length();
```

clang-tidy and other static analyzers complain because the first assignment is pointless. Instead we check to see if we would fall into that second block and skip the first initialization if it would.

Reply:
I pattern-matched too narrowly :-) — I get it now. Thanks for the cluebat.